HW03/ekaterina_balabanovich/spec/balabanovich_hw03_task_3_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require_relative '../balabanovich_hw03_task_3'
+
+RSpec.describe 'Task3' do
+  let(:logs) do
+    <<~LOGS
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - docker event processed
+      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - Calling core with action: messages
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
+    LOGS
+  end
+
+  describe '#extract_core_log' do
+    subject { extract_core_log(logs) }
+
+    context 'when logs are not empty' do
+      let(:expected_output) do
```

Comment:
When context is empty it looks strange a little bit, I think it will be ok to duplicate this `logs` to make it more readable or write something like this `let(:logs) { base_logs }`
```suggestion
    context 'when logs are not empty' do
   let(:logs) do
    <<~LOGS
      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - docker event processed
      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - Calling core with action: messages
      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
    LOGS
  end
      let(:expected_output) do
```